docs/how_to_guides/how_to_run_models_in_a_py_script.rst

```diff
@@ -90,32 +90,32 @@ Example: Python file with recommended structure
        m.fs.unit.A_comp.fix(4.2e-12)  # membrane water permeability
        m.fs.unit.B_comp.fix(3.5e-8)  # membrane salt permeability
        m.fs.unit.permeate.pressure[0].fix(101325)  # permeate pressure
-       
+
        # Set scaling factors for component mass flowrates.
        m.fs.properties.set_default_scaling('flow_mass_phase_comp', 1, index=('Liq', 'H2O'))
        m.fs.properties.set_default_scaling('flow_mass_phase_comp', 1e2, index=('Liq', 'NaCl'))
-       
+
        # Calculate scaling factors.
        calculate_scaling_factors(m)
 
-       # Get default watertap solver 
+       # Get default watertap solver
        solver = get_solver()
-       
-       # Initialize the model passing default solver options 
+
+       # Initialize the model passing default solver options
        m.fs.unit.initialize(optarg=solver.options)
-       
+
        # Solve the model (using the tee=True option to display solver info)
-       results = solver.solve(m, tee=True)
+       solver.solve(m, tee=True)
+
 
-   # Call that function in the "__main__" for the script 
+   # Call that function in the "__main__" for the script
    if __name__ == "__main__":
        main()
 
```

Comment:
```suggestion
.. testcleanup::

   # run `main`
   main()

```

This way `main` get executed as part of the test, and not just checked for syntax.

Reply:
This gives an error about `NameError: name 'main' is not defined`.

Reply:
Yeah. That's because I misremembered above and we should be using the `.. testcode::` (and *not* doctest) directive for the main blocks.

However, this seems to be a pervasive issue in the documentation which should be addressed separately. Therefore, I suggest you revert 08a1d8c (as it compounds this issue) and I'll deal with that separately.

This code doesn't even work as-is (with your correction), so we need to fix it and probably a lot of other examples.

Reply:
I used the `.. testcode::` directive for the corrected example and used `.. code-block::` for the example that may cause errors on Windows. With these changes everything is working fine.

Reply:
Yes, that will test fine for now. Of course, `main` is still not executed, so we're not seeing the errors from that.

But that is okay, I will try to address this later as part of another PR.